lib/kafka/compression.rb

```diff
@@ -0,0 +1,23 @@
+require "kafka/snappy_codec"
+require "kafka/gzip_codec"
+
+module Kafka
+  module Compression
+    def self.find_codec(name)
+      case name
+      when nil then nil
+      when :snappy then SnappyCodec.new
+      when :gzip then GzipCodec.new
```

Comment:
totally disagree (but its your gem ;) ) - having ids, names and classes defined in one place makes it easier to maintain and develop